hoomd/md/ForceComposite.cc

```diff
@@ -589,20 +591,19 @@ void ForceComposite::createRigidBodies()
                 // Position and orientation are handled by updateCompositeParticles.
                 snap.type[constituent_particle_tag]
                     = h_body_type.data[m_body_idx(body_type, current_body_index)];
-                snap.body[constituent_particle_tag] = central_particle_tag;
+                snap.body[constituent_particle_tag] = particle_tag;
                 snap.charge[constituent_particle_tag]
                     = m_body_charge[body_type][current_body_index];
                 snap.diameter[constituent_particle_tag]
                     = m_body_diameter[body_type][current_body_index];
-                snap.pos[constituent_particle_tag] = snap.pos[central_particle_tag];
+                snap.pos[constituent_particle_tag] = snap.pos[particle_tag];
 
                 // Since the central particle tags here will be [0, n_central_particles), we know
                 // that the molecule number will be the same as the central particle tag.
-                molecule_tag[constituent_particle_tag] = central_particle_tag;
+                molecule_tag[constituent_particle_tag] = particle_tag;
 
                 ++constituent_particle_tag;
```

Comment:
@tommy-waltmann The issue was that `create_bodies` was setting the body tag of the particles incorrectly which led to rigid body validation errors at the start of the simulation. These changes here fix it to set `body[i] = i` for those particles that are the central particles of rigid bodies.